### PR TITLE
Create instance from worker

### PR DESCRIPTION
**Pull request by JolifantoBambla**
**Checklist**

- [x] Run `cargo clippy`.
- [x] Run `RUSTFLAGS=--cfg=web_sys_unstable_apis cargo clippy --target wasm32-unknown-unknown` if applicable.
- [x] Add change to CHANGELOG.md. See simple instructions inside file.

**Connections**
https://github.com/gfx-rs/wgpu/issues/1986
https://github.com/gfx-rs/wgpu/issues/1466
https://github.com/gfx-rs/wgpu/issues/2652

**Description**
On the WebGPU backend, `Context::init()` uses `web_sys::window()` to access `navigator.gpu`.
Since `window` is not available on web workers this function panics when called from a web worker (#1986).
This is solved by introducing a `Global` type the result of `js_sys::global()` is cast to.
This instance of `Global` is then used to check whether or not `window()` is defined to determine if the function was called in a worker context or not. If it's in a worker context, it will use `WorkerGlobalScope` instead of `Window` to access the `gpu` property.
I'm relatively new to rust and took the solution from [here](https://github.com/rustwasm/gloo/blob/2c9e776701ecb90c53e62dec1abd19c2b70e47c7/crates/timers/src/callback.rs#L8-L40) after stumbling upon [this issue in wasm-bindgen](https://github.com/rustwasm/wasm-bindgen/issues/2148), so it's maybe not the prettiest solution.
Let me know if you have a better solution in mind.

**Testing**
I added logging of `adapter.get_info()` to the `hello` example and am about to submit another PR over at https://github.com/gfx-rs/wgpu-rs.github.io that adds an example running the `hello` example in a webworker.
Note though, that this example doesn't work in Firefox since it doesn't support loading modules in workers. I tested it in Chromium on Linux.

### CODE REVIEW

**Comment by JolifantoBambla** [comment]
the example I used to test this is here: https://github.com/gfx-rs/wgpu-rs.github.io/pull/24

**Comment by JolifantoBambla** [comment]
Hey, @cwfitzgerald, I fixed the formatting. Could you maybe try running the checks again? Thanks :)

File: wgpu/examples/hello/main.rs [hunk 1]
```diff
@@ -16,6 +16,9 @@ async fn run() {
             .unwrap()
     };
 
+    #[cfg(target_arch = "wasm32")]
+    log::info!("{:?}", adapter.get_info());
```
Comment (cwfitzgerald) [inline wgpu/examples/hello/main.rs:20]: I feel like we should either always use log - or explain why we're only sometimes using it.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
One comment, but otherwise looks good.

File: wgpu/examples/hello/main.rs (hunk 1 above)
Comment (JolifantoBambla) [reply wgpu/examples/hello/main.rs]: Good point! I changed it to use log now in all cases. Or is there a reason log wasn't used in the non-wasm case before?

**Review by cwfitzgerald — APPROVED** [review]
Nice!

**Pull request merged**